platform_plugin_aspects/utils.py

```diff
@@ -49,12 +51,15 @@ def generate_superset_context(
     course_id = context["course_id"]
     superset_config = settings.SUPERSET_CONFIG
 
+    # We're modifying this, keep a local copy
+    rtn_dashboards = copy.deepcopy(dashboards)
+
     if language:
-        for dashboard in dashboards:
+        for dashboard in rtn_dashboards:
             if not dashboard.get("allow_translations"):
                 continue
             dashboard["slug"] = f"{dashboard['slug']}-{language}"
-            dashboard["uuid"] = str(get_uuid5(dashboard["uuid"], language))
+            dashboard["uuid"] = get_localized_uuid(dashboard["uuid"], language)
```

Comment:
We should create a new dashboard list and generate all UUIDs here so that we have a better separation of concerns.

Reply:
I'm not sure what you mean, this is already operating on a copy of the dashboard list?